### ISSUE TITLE
warnings in development and production

### ISSUE THREAD

**Issue by sayduck-daniel** [ForestAdmin/forest-rails#271]
I'm using ruby 2.5.1 and it seems I get new warnings I didn't have before. Any idea how to get ride of those?

```
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: already initialized constant ForestLiana::UserSpace::CustomerSerializer
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: previous definition of CustomerSerializer was here
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: already initialized constant ForestLiana::UserSpace::MapSerializer
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: previous definition of MapSerializer was here
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: already initialized constant ForestLiana::UserSpace::ModelSerializer
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: previous definition of ModelSerializer was here
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: already initialized constant ForestLiana::UserSpace::ReviewChannelSerializer
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: previous definition of ReviewChannelSerializer was here
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: already initialized constant ForestLiana::UserSpace::SharedAssetSerializer
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: previous definition of SharedAssetSerializer was here
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: already initialized constant ForestLiana::UserSpace::UserSerializer
/Users/daniel/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.8.2/app/serializers/forest_liana/serializer_factory.rb:11: warning: previous definition of UserSerializer was here```

**Comment by tvilon** [comment]
I have the same problem

**Comment by arnaudbesnier** [comment]
Hey @sayduck-daniel, thanks for the report. Did those warning appear since a specific Forest liana version? Or did they appear after the ruby version upgrade on your projects?

Thanks for the help.

**Comment by sayduck-daniel** [comment]
I can’t recall if it was during an upgrade of forest or not. I’m using 2.5.1 ruby.

**Comment by tvilon** [comment]
For us it happened when we added smart action on the 3rd collection. Originally we had smart actions only on `User` and `Organization` and it worked fine.

Then we added the skeleton for a new smart action on `UserReview` and that generated the warning. I say skeleton because indeed you only need to create the `/lib/forest_liana/collections/xxxx.rb` to have the warning. 
```ruby
# /lib/forest_liana/collections/user_review.rb
class Forest::UserReview
  include ForestLiana::Collection

  collection :UserReview

end
```

**Findings**

The warning fires super early in the rais boot. Even before rails checks if another server is already running.

It looks like `ForestLiana::UserSpace::UserReviewSerializer` can be defined twice 

-`https://github.com/ForestAdmin/forest-rails/blob/208b987e3081feea2202fff02e66fa314de611a1/lib/forest_liana/collection.rb#L17`
-`https://github.com/ForestAdmin/forest-rails/blob/be30d86774c04552059e7c32d1c27452729f4ab4/app/serializers/forest_liana/serializer_factory.rb#L11`

**Comment by melcher** [comment]
Getting the same error with 2.13.7:
```
/Users/user/gems/forest_liana-2.13.7/app/serializers/forest_liana/serializer_factory.rb:11: warning: already initialized constant ForestLiana::UserSpace::CompanySerializer
/Users/user/gems/forest_liana-2.13.7/app/serializers/forest_liana/serializer_factory.rb:11: warning: previous definition of CompanySerializer was here
/Users/user/gems/forest_liana-2.13.7/app/serializers/forest_liana/serializer_factory.rb:11: warning: already initialized constant ForestLiana::UserSpace::InvoiceSerializer
/Users/user/gems/forest_liana-2.13.7/app/serializers/forest_liana/serializer_factory.rb:11: warning: previous definition of InvoiceSerializer was here
/Users/user/gems/forest_liana-2.13.7/app/serializers/forest_liana/serializer_factory.rb:11: warning: already initialized constant ForestLiana::UserSpace::ReceiptSerializer
/Users/user/gems/forest_liana-2.13.7/app/serializers/forest_liana/serializer_factory.rb:11: warning: previous definition of ReceiptSerializer was here
/Users/user/gems/forest_liana-2.13.7/app/serializers/forest_liana/serializer_factory.rb:11: warning: already initialized constant ForestLiana::UserSpace::PartnerSerializer
/Users/user/gems/forest_liana-2.13.7/app/serializers/forest_liana/serializer_factory.rb:11: warning: previous definition of PartnerSerializer was here
```

That's just a snippet, we get about 20 of these errors on every rails boot / env load. We treat messages printed at load to be warnings / errors that need to be addressed but have trouble seeing them given the amount of text that now appears on load. Would be great if this could be addressed.

**Comment by KidA001** [comment]
I also started getting this error on forest_liana `2.13.6` once I added a custom field

```ruby
# frozen_string_literal: true

class Forest::Company
  include ForestLiana::Collection

  collection :Company

  field :logo, type: 'String' do
    object.try(:logo).try(:service_url)
  end
end

```

**Comment by sayduck-daniel** [comment]
Awesome track down and f the issue. It appears I load lib as well.

Let me know if you found a work around to exclude the folder

**Comment by arnaudbesnier** [comment]
Fixed in v4.0.1 @sayduck-daniel ?

**Comment by sayduck-daniel** [comment]
Damn I forget about this issue... Yes indeed, it's fixed
